fibreda/readme.md

```diff
@@ -0,0 +1,512 @@
+
+### 8.4 Get pathway
+
+* Accept `bitmap` or `indices`. If neither provided, return **all** stored rows for the commitment.
+* Partial fulfillment allowed; include `missing_indices`.
+
+### 8.5 Promotion Listener
+
+* Subscribe to on-chain **PFB** events.
+* On observed `commitment`, call `Promote()` to move from `bu/*` to `bc/*` and update pointer `p/*`.
+* Optionally **issue a new attestation** for the confirmed period and expose via `GetAttestation(Commitment)` or piggyback on a promotion ack.
+
+### 8.6 API limits & fairness
+
+* Enforce `server_rows_per_message_limit` and maximum serialized request size.
+* **Rate-limit** ingress to **\~10 MiB/s** (configurable). With \~**300 KiB**/req worst-case, this is **\~33 RPS**. This is **conservative** because not all requests are max size; start here and raise with telemetry.
+* Include `backoff_ms` in responses to guide clients under pressure.
+
+### 8.7 Storage roadmap
+
+* **SQL deletes are not relevant in v1**. We **start with Badger**. Later we may prototype **SQL-light** and **epoch-bucket** stores and pick based on measured throughput and GC performance.
+
+### 8.8 Operational notes
+
+* Use Badger write batches; ACK only after `Flush()+Sync`.
+* Value‑log GC after sweeps or periodically until no‑rewrite.
+* Health endpoints: `/healthz` (process up), `/readyz` (DB open, GC loop running), metrics endpoint (Prometheus).
+
+---
+
+## 9) Security considerations
+
+* **Attestation domain** prevents cross‑protocol replay; `chain_id` prevents cross‑network reuse; `service_epoch_minute` binds to a freshness window.
+* **Consensus keys** (ed25519) are the identity; we do not bind endpoint/IP in v1. If needed later, add a nonce or endpoint hash in v2.
+* **Row verification** includes Context‑Based Verification which is necessary to prevent malleability via valid‑looking but non‑decodable rows.
+
+---
+
+## 10) Observability
+
+**Client metrics**: encode latency, rows count, bytes per FSP, upload latency per FSP, attestation verify failures, quorum time, PFB submit latency/height, refresh PFB count.
+
+**Server metrics**: RPS, bytes/s, dedup hit rate, proof verify failures, RLC failures, write batch size, GC sweep durations, promotion count, `backoff_ms` emissions, Badger vlog GC stats.
+
+**Logs**: structured; include commitment (hex32), validator addr, service\_epoch\_minute, request sizes, error codes.
+
+---
+
+## 11) Open questions & options
+
+### 11.1 Replication factor & mapping
+
+Let `K` = originals, `ρ` = parity ratio (`encoding_factor`), `R` = replication factor, `V` = validators.
+
+Target rows per validator:
+
+```
+rows_per_val = ceil( (K * (1 + ρ) * R) / V )
+```
+
+Assignment options:
+
+* **Deterministic R**: for each row, take top‑`R` lowest SipHash scores over validators.
+* **Probabilistic**: choose a cutoff so expected replicas ≈ `R`.
+
+Impact: quorum guarantees under `f < V/3` faulty validators; pick `R` to ensure reconstruction margin.
```

Comment:
might be useful to keep this info all in one place as I'm a bit confused as to what is the open question relative to here and a different version of it is in the ### ShardMap details section